android-components/components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -72,6 +72,12 @@ object InitAction : BrowserAction()
  */
 object RestoreCompleteAction : BrowserAction()
 
+/**
+ * [BrowserAction] implementation for updating state related to whether the extensions process
+ * spawning has been disabled and a popup is necessary
+ */
+class ExtensionProcessDisabledPopupAction(val showPopup: Boolean): BrowserAction()
```

Comment:
We prefer `data class`es for actions for the derived `copy()`, because they generally suggest "simpler" types, and because they slightly restrict the design space for actions.

Reply:
ya, that definitely makes the most sense. Good catch, I've updated it.